controllers/rule_controller.go

```diff
@@ -57,3 +107,49 @@ func (r *RuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&apiv1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *RuleReconciler) updateRulesConfigmap(ctx context.Context, data string) error {
+
+	var oathkeeperRulesConfigmap apiv1.ConfigMap
+
+	if err := r.Get(ctx, r.RuleConfigmap, &oathkeeperRulesConfigmap); err != nil {
+		if apierrs.IsNotFound(err) {
+
+			oathkeeperRulesConfigmap = apiv1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      r.RuleConfigmap.Name,
+					Namespace: r.RuleConfigmap.Namespace,
+				},
+				Data: map[string]string{"rules": data},
+			}
+
+			err := retry.Do(func() error {
+				return r.Create(ctx, &oathkeeperRulesConfigmap)
+			},
+				retry.Attempts(5),
+				retry.DelayType(retry.BackOffDelay))
+
+			if err != nil {
+				r.Log.Error(err, "unable to create configmap")
+				os.Exit(1)
+			}
+
+			return nil
+		}
+
+		return err
+	}
+
+	oathkeeperRulesConfigmapCopy := oathkeeperRulesConfigmap.DeepCopy()
```

Comment:
Do we need to make a deepcopy? What will happen if we just put `oathkeeperRulesConfigmap.Data = map[string]string{"rules": data}` ?

Reply:
Resolved with commit `2989ca9`